pkg/operator/controllers/guardrails/policies/README.md

```diff
@@ -0,0 +1,305 @@
+        kinds: ["Job", "CronJob"]
+      - apiGroups: ["rbac.authorization.k8s.io"]
+        kinds: ["Role", "RoleBinding"]
+      - apiGroups: ["policy"]
+        kinds: ["PodDisruptionBudget"]
+```
+
+## Test the rego
+
+* install opa cli, refer https://github.com/open-policy-agent/opa/releases/
+
+* after _test.go is done, test it out, and fix the problem
+  ```sh
+  opa test ../library/common.rego *.rego [-v] #-v for verbose
+  ```
+
+## Generate the Constraint Templates
+
+* install gomplate which is used by generate.sh, see https://docs.gomplate.ca/installing/
+
+* execute generate.sh under policies, which will generate the acutal Constraint Templates under gktemplates folder, example:
+
+  ```sh
+  ARO-RP/pkg/operator/controllers/guardrails/policies$ ./scripts/generate.sh 
+  Generating gktemplates/aro-deny-delete.yaml from gktemplates-src/aro-deny-delete/aro-deny-delete.tmpl
+  Generating gktemplates/aro-deny-privileged-namespace.yaml from gktemplates-src/aro-deny-privileged-namespace/aro-deny-privileged-namespace.tmpl
+  Generating gktemplates/aro-deny-labels.yaml from gktemplates-src/aro-deny-labels/aro-deny-labels.tmpl
+  ```
+
+## gator test
+
+Create suite.yaml and testcases in gator-test folder under the folder created for the new policy, refer example below:
+
+```yaml
+kind: Suite
+apiVersion: test.gatekeeper.sh/v1alpha1
+metadata:
+  name: privileged-namespace
+tests:
+- name: privileged-namespace
+  template: ../../gktemplates/aro-deny-privileged-namespace.yaml
+  constraint: ../../gkconstraints-test/aro-priv-ns-operations.yaml
+  cases:
+  - name: ns-allowed-pod
+    object: gator-test/ns_allowed_pod.yaml
+    assertions:
+    - violations: no
+  - name: ns-disallowed-pod
+    object: gator-test/ns_disallowed_pod.yaml
+    assertions:
+    - violations: yes
+      message: User test-user not allowed to operate in namespace openshift-config
+  - name: ns-disallowed-deploy
+    object: gator-test/ns_disallowed_deploy.yaml
+    assertions:
+    - violations: yes
+      message: User test-user not allowed to operate in namespace openshift-config
+```
+gkconstraints-test here stores the target yaml files after expanding "{{.Enforcement}}" symbol.
+
+gator tests ConstraintTemplate and Constraint together, items under cases keyword are test cases indicator, everyone pointing to a yaml file in gator-test, which provides test input for one scenario, example:
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: allowed
+  namespace: test
+spec:
+  serviceAccountName: test-user
+  containers:
+    - name: test
+      image: openpolicyagent/opa:0.9.2
+      args:
+        - "run"
+        - "--server"
+        - "--addr=localhost:8080"
+      resources:
+        limits:
+          cpu: "100m"
+          memory: "30Mi"
+```
+the assertions section is the expected result
+
+gator test is done via cmd:
+
+test.sh executes both opa test and gator verify
+```sh
+ARO-RP/pkg/operator/controllers/guardrails/policies$ ./scripts/test.sh
```

Comment:
more `make` opportunities, I'm going to stop here - I'd argue every shell command in this README section should somehow be part of a single `make` flow.